### PR TITLE
Create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by viet-nv**

### CODE REVIEW

**Comment by viet-nv** [comment]
@nguyenhoaidanh Can you take a look at issue number 3 and number 4
https://github.com/KyberNetwork/dmm-interface/security/code-scanning/3
https://github.com/KyberNetwork/dmm-interface/security/code-scanning/4

I need you on this then we can merge this PR

**Comment by nguyenhoaidanh** [comment]
@viet-nv  updated.

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "main", develop, staging ]
```
Comment (namgold) [inline .github/workflows/codeql-analysis.yml:16]: why `"main"` is so special

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'javascript' ]
```
Comment (namgold) [inline .github/workflows/codeql-analysis.yml:35]: it [supported](https://aka.ms/codeql-docs/language-support) `typescript`

**Comment by nguyenhoaidanh** [comment]
I saw duplicate action when I push 1 commit. 
image

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (viet-nv) [reply .github/workflows/codeql-analysis.yml:16]: this is auto generated from codeQL template. haha

File: .github/workflows/codeql-analysis.yml (hunk 2 above)
Comment (viet-nv) [reply .github/workflows/codeql-analysis.yml:35]: typescript is trigger by default, you can see the log

**Comment by viet-nv** [comment]
> I saw duplicate action when I push 1 commit. image

there are 2 different action, one on issue number 3, one on new commit

File: .github/workflows/codeql-analysis.yml (hunk 2 above)
Comment (namgold) [reply .github/workflows/codeql-analysis.yml:35]: Tin chuẩn chưa anh
239997824_1673803459488529_3231250185561427304_n

Comment (viet-nv) [reply .github/workflows/codeql-analysis.yml:35]: definitely "Chuẩn"

**Review by namgold — APPROVED** [review]

**Pull request merged**